### PR TITLE
Fix/post request

### PR DESCRIPTION
**Pull request by FrazierLE**
Summary of the work done:

- This completes post requests for adding and removing ingredients
- Brings functionality to cook recipe button
- Displays button or missing ingredients depending on pantry items

Contributed to code:

- Name Eleanor and Lauren 

Needs to review:

- Name Kiko and Jenn

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -162,42 +164,43 @@ function displayRecipeDetailPage(event) {
...
+    return user.pantry.ingredientsNeeded.map((ingredientNeed)=>{
+        let ingredientName = ingredients.ingredients.reduce((name, ingredient)=>{
+            if (ingredientNeed.missingIngredient === ingredient.id) {
+             name = ingredient.name
+            }
+            return name
+         }, "")
+         return {name: ingredientName, quantity: ingredientNeed.quantityNeeded, unit:ingredientNeed.units} 
+     }).forEach((missing)=>{
+        missingIngredientList.innerHTML += `<li>${missing.quantity} ${missing.unit} ${missing.name}</li>`
```
Comment (knpausch) [inline src/scripts.js:200]: could we format the missing.quantity with toFixed(2). When a recipe is missing 1/3 cup of something, it displays it as 0.33333333333333333333333

Comment (Eleanorgruth) [reply src/scripts.js:200]: That's a great catch Kiko! I think that we also should add `.toLowerCase()` to the units, the data is inconsistent and sometimes the unit is in caps.

**Review by jmyacobn — APPROVED** [review]
Functionality works great! I would only suggest a couple of minor things for user experience. 
- Maybe letting the user know that something happened when they clicked the cook button. A small message that says, "Items used for recipe now removed from your pantry."
- Clearing out the form fields after each addition, again so the user knows something happened. I found myself adding somethings multiple times because I was unsure.

File: src/scripts.js [hunk 2]
```diff
@@ -65,7 +67,7 @@ removeRecipeButton.addEventListener('click', removeRecipeFromFavorites)
 favoriteButton.addEventListener('click', displayFavoritesPage)
 pantryButton.addEventListener('click', displayPantryPage)
 addButton.addEventListener('click', addItemToPantry)
-cookRecipeButton.addEventListener('click', removeIngredientsFromPantry)
+cookRecipeButton.addEventListener('click', cookRecipe)
 searchBar.addEventListener('keypress', (event) => {
     if (event.key === "Enter" && homeView) {
```
Comment (knpausch) [inline src/scripts.js:72]: FYI when user is on pantry page, enters quantity and presses "Enter" it shoots them back to the home screen. We can add conditional here to check if we're on home view

**Pull request merged**